src/util/urlUtils.js

```diff
@@ -0,0 +1,11 @@
+export default function getParameterByName(name) {
+  const url = window.location.href;
+  name = name.replace(/[[]]/g, "\\$&");
+  var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+      results = regex.exec(url);
+  if (!results) return null;
+  if (!results[2]) return '';
+  if (results[2] === 'false') return false;
+  if (results[2] === 'true') return true;
```

Comment:
I moved this function from `App.js` and added parsing for `boolean` values. I'm not sure if there's a cleaner/more canonical way to do this.

Reply:
Looks good.

Reply:
...as I'm refactoring, I realize I never quite understood the reason for putting the url parsing in the ProteinViewer code.

I feel like this should be part of App. The url is setting what the App's state should be, which then passes it off to "dumb" components that have no idea about url parameters. When this is embedded in another app (MLM, for instance), that app should also get to control what the props of the components it's showing are, without forcing the use of specific url parameters in that app.

Thoughts?

Reply:
Yeah I think that makes sense! Having the App configure everything does seem cleaner and more usable moving forward